frontend/pages/create-edit-user.js

```diff
@@ -0,0 +1,367 @@
+            variables: {
+              request: { ethereumAddress: props.address },
+            },
+          }),
+          currentProfile = getLensUser.data.defaultProfile;
+        if (currentProfile !== null) {
+          setFormValues({
+            ...defaultFormValues,
+            handle: currentProfile.handle,
+            bio: currentProfile.bio,
+            name: currentProfile.name,
+            ...Object.fromEntries(
+              currentProfile.attributes.map((field) => {
+                return [field.key, field?.value || ""];
+              })
+            ),
+          });
+        }
+        // Setting a default profile is not working for some reason
+        setLensHandle(currentProfile?.handle);
+      } catch (err) {
+        setHasIPFonts(""), setLensHandle("");
+      }
+    })();
+  }, [props.address]);
+
+  // Create/edit user function
+  async function handleCreateEditUser(body, resetFunction) {
+    setWaitForm(true);
+    // Creates a test lens profile
+    const getLensUser = await lensClient.query({
+      query: getDefaultProfile,
+      variables: { request: { ethereumAddress: props.address } },
+    });
+    if (!getLensUser.data.defaultProfile?.handle) {
+      try {
+        const createLensProfile = await lensClient.mutate({
+          mutation: createProfile,
+          variables: {
+            request: {
+              handle: body.handle,
+              profilePictureUri: null,
+              followModule: { freeFollowModule: true },
+              followNFTURI: null,
+            },
+          },
+        });
+        if (createLensProfile.data.createProfile.reason === "HANDLE_TAKEN") {
+          alert("Lens handle taken, change it and try again!");
+          setWaitForm(false);
+          return;
+        }
+        if (createLensProfile.data.createProfile.txHash) {
+          const profileId = await lensClient.query({
+              query: getProfileByAddress,
+              variables: { owner: props.address },
+            }),
+            profiles = profileId.data.profiles.items,
+            setProfile = await lensClient.mutate({
+              mutation: createSetDefaultProfile,
+              variables: {
+                request: {
+                  profileId: profiles[profiles.length - 1].id,
+                },
+              },
+            });
+          if (
+            setProfile.data.createSetDefaultProfileTypedData.typedData.value
+              .profileId
+          ) {
+            setLensHandle(`${body.handle}.test`);
+            resetFunction();
+          }
+        }
+      } catch (err) {
+        alert(`There was an error creating your lens test profile ${err}`);
+        resetFunction();
+        return;
+      }
+    }
+    try {
+      const ipfontsContract = await connectContract();
+      if (ipfontsContract) {
+        const ipfontsBody = {
+            email: body?.email,
+            name: body?.name,
+            website: body?.website,
+            bio: body?.bio,
```

Comment:
Can body be null ( e.g. `body?` ) ? The yup validator library should take care of validating the form and not allowing this code to run no?

Reply:
I think this approach was after starting everything as string and I didn't change it, I need to check that, thanks for pointing it JP :)